[docs] Improve AppAuth docs (#6876)
expo/expo: docs/pages/versions/unversioned/sdk/app-auth.md

sourceCodeUrl: 'https://github.com/expo/expo/tree/sdk-36/packages/expo-app-auth'
---

import SnackInline from '~/components/plugins/SnackInline';
import TableOfContentSection from '~/components/plugins/TableOfContentSection';

**`expo-app-auth`** allows you to authenticate and authorize your users through the native OAuth library AppAuth by [OpenID](https://github.com/openid).

Many services that let you authenticate with them or login with them, like GitHub, Google, GitLab, etc., use the OAuth 2.0 protocol. It's the industry standard.

If you are trying to implement sign in with [Google](../google-sign-in) or [Facebook](../facebook), there are special modules in the Expo SDK for those (though this module will work).

> For now, this module only works on Anroid and iOS, but web support is possible is planned to be added.

Reviewer: ```suggestion
Currently, this module only supports on Android, and iOS. Web support is planned to be added. Track it here: [Web support in expo-app-auth](https://github.com/expo/expo/issues/6883).
```

> Below is a set of example functions that demonstrate how to use `expo-app-auth` with the Google OAuth Sign-In provider.

Reviewer: ```suggestion
Below is a set of example functions that demonstrate how to use `expo-app-auth` with the Google OAuth sign in provider.
```

<SnackInline>

> ```js

Reviewer: ```suggestion
```tsx
```

better colors

Author: cool

import React, { useEffect, useState } from 'react';
import { AsyncStorage, Button, StyleSheet, Text, View } from 'react-native';
import * as AppAuth from 'expo-app-auth';

export default function App() {
  let [authState, setAuthState] = useState(null);

  useEffect(() => {
    (async () => {
      let cachedAuth = await getCachedAuthAsync();
      if (cachedAuth && !authState) {
        setAuthState(cachedAuth);
      }
    })();
  }, []);

  return (
    <View style={styles.container}>
      <Text>Expo AppAuth Example</Text>

>       <Text></Text>

Reviewer: ```suggestion

```

Author: I think I was just adding whitespace here but there's probably a better way to do it and this looks sloppy.

      <Button
        title="Sign In with Google "
        onPress={async () => {

>           let _authState = await signInAsync();

Reviewer: ```suggestion
          const authState = await signInAsync();
```
no underscore

Author: same as other comments about `let` and `const`

          setAuthState(_authState);
        }}
      />
      <Button
        title="Sign Out "
        onPress={async () => {
          await signOutAsync(authState);
          setAuthState(null);
        }}
      />

>       <Text>{JSON.stringify(authState)}</Text>

Reviewer: ```suggestion
      <Text>{JSON.stringify(authState, null, 2)}</Text>
```

human-readable format for the JSON.

    </View>
  );
}

> let styles = StyleSheet.create({

Reviewer: ```suggestion
const styles = StyleSheet.create({
```

Author: I think `const` is overused in most circumstances. 
https://overreacted.io/on-let-vs-const/

  return null;
}

function checkIfTokenExpired({ accessTokenExpirationDate }) {
  return new Date(accessTokenExpirationDate) < new Date();
}

async function refreshAuthAsync({ refreshToken }) {

>   let authState = await AppAuth.refreshAsync(config, refreshToken);

Reviewer: why `let`?

Author: I just use `let` everywhere unless I want to strongly indicated intent to make something `const`.

Similar reasons as described here. https://overreacted.io/on-let-vs-const/
